src/data/complex/basic.lean

```diff
@@ -196,35 +196,41 @@ instance : star_ring ℂ :=
 /-! ### Norm squared -/
 
 /-- The norm squared function. -/
-@[pp_nodot] def norm_sq (z : ℂ) : ℝ := z.re * z.re + z.im * z.im
+@[pp_nodot] def norm_sq : monoid_with_zero_hom ℂ ℝ :=
+{ to_fun := λ z, z.re * z.re + z.im * z.im,
+  map_zero' := by simp,
+  map_one' := by simp,
+  map_mul' := λ z w, by { dsimp, ring } }
+
+lemma norm_sq_def (z : ℂ) : norm_sq z = z.re * z.re + z.im * z.im := rfl
```

Comment:
Is `norm_sq_apply` a better name, to match the names that `simps` would generate?

Reply:
Yes, it is.
```suggestion
lemma norm_sq_apply (z : ℂ) : norm_sq z = z.re * z.re + z.im * z.im := rfl
```